ext/intl/dateformat/dateformat_parse.c

```diff
@@ -149,14 +158,53 @@ PHP_FUNCTION(datefmt_parse)
 			RETURN_FALSE;
 		}
 	}
-	internal_parse_to_timestamp( dfo, text_to_parse, text_len, z_parse_pos?&parse_pos:NULL, return_value);
+	internal_parse_to_timestamp( dfo, text_to_parse, text_len, z_parse_pos?&parse_pos:NULL, false, return_value);
 	if(z_parse_pos) {
 		zval_ptr_dtor(z_parse_pos);
 		ZVAL_LONG(z_parse_pos, parse_pos);
 	}
 }
 /* }}} */
 
+PHP_METHOD(IntlDateFormatter, parseToCalendar)
+{
+	char*           text_to_parse = NULL;
+	size_t          text_len =0;
+	zval*         	z_parse_pos = NULL;
+	int32_t		parse_pos = -1;
+
+	DATE_FORMAT_METHOD_INIT_VARS;
+
+	/* Parse parameters. */
+	if( zend_parse_method_parameters( ZEND_NUM_ARGS(), getThis(), "Os|z!",
```

Comment:
You don't need to use `zend_parse_method_parameters()` when just adding a new method, as the `getThis()` is pointless and the first param being an object is also unnecessary.